### PR TITLE
Add docs for version 2.0

### PR DESCRIPTION
**Pull request by fabianfett**
- Removed overly verbose issue and pr templates
- Removed documentation from README
- Readme now links to spi documentation page
- Added docc documentation
- Removed last usages of `TimeHistogram`

### CODE REVIEW

File: Sources/Prometheus/Docs.docc/index.md [hunk 1]
```diff
@@ -5,15 +5,77 @@ A prometheus client library for Swift.
...
 
 ``Prometheus`` supports creating ``Counter``s, ``Gauge``s and ``Histogram``s and exporting their
-values in the Prometheus export format. 
+values in the Prometheus text format.
+
+``Prometheus`` integrates with [Swift Metrics](doc:swift-metrics).
+
+## Installation
+
+``Prometheus`` is available through Swift Package Manager. To include it in your project add the
```
Comment (FranzBusch) [inline Sources/Prometheus/Docs.docc/index.md:14]:
```suggestion
``Prometheus`` is available through the Swift Package Manager. To include it in your project add the 
```

File: Sources/Prometheus/Docs.docc/swift-metrics.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      .product(name: "Prometheus", package: "swift-prometheus"),
+      .product(name: "Metrics", package: "swift-metrics"),
+    ]
+  ),
+```
+
+### Setting up Prometheus to export to Swift Metrics
+
+At application startup you need to create a ``PrometheusMetricsFactory`` to register it as
+metrics backend for swift-metrics:
```
Comment (FranzBusch) [inline Sources/Prometheus/Docs.docc/swift-metrics.md:40]:
```suggestion
the metrics backend for swift-metrics:
```

File: Sources/Prometheus/Docs.docc/swift-metrics.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+```
+
+Now a `Timer` with the label "long" will use the buckets  
+`[.seconds(5), .seconds(10), .seconds(25), .seconds(50), .seconds(100),]`, whereas any other 
+`Timer` will use the default buckets 
+`[.milliseconds(5), .milliseconds(10), .milliseconds(25), .milliseconds(50), .milliseconds(100),]`.
+
+The same functionality is also available for ``ValueHistogram`` and aggregating `Recorder`s.
+
+[Swift Metrics]: https://github.com/apple/swift-metrics
```
Comment (FranzBusch) [inline Sources/Prometheus/Docs.docc/swift-metrics.md:170]: Why this link here at the bottom?

**Review by FranzBusch — APPROVED** [review]
Looks good. Just a couple of small nits

File: Sources/Prometheus/Docs.docc/index.md (hunk 1 above)
Comment (fabianfett) [reply Sources/Prometheus/Docs.docc/index.md:14]: Not sure "the" is needed here.

**Review by ktoso — APPROVED** [review]
Looks ok, thx

File: Sources/Prometheus/Docs.docc/swift-metrics.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+This can be particularly usefull, if you want to change the names and labels for metrics that are
+generated in a third party library.
+
+### Defining Buckets for Histograms
+
+#### Default buckets
+
+Swift Metric ``Timer``s are backed by a Promtheus ``DurationHistogram`` and Swift Metric 
+``Recorder``s, that aggregate are backed by a Prometheus ``ValueHistogram``. As a user, you can 
+specify which buckets shall be used within the backing ``Histogram``s.
```
Comment (hamzahrmalik) [inline Sources/Prometheus/Docs.docc/swift-metrics.md:111]:
```suggestion
Swift Metric ``Timer``s are backed by a Prometheus ``DurationHistogram`` and Swift Metric 
``Recorder``s that aggregate are backed by a Prometheus ``ValueHistogram``. As a user, you can 
specify which buckets shall be used within the backing ``Histogram``s.
```

File: Sources/Prometheus/PrometheusMetricsFactory.swift [hunk 5]
```diff
@@ -43,15 +43,15 @@ public struct PrometheusMetricsFactory: Sendable {
     /// The histogram buckets for a ``ValueHistogram`` per label
     public var valueHistogramBuckets: [String: [Double]]
 
-    /// A closure to modify the label and dimension names used in the Swift Metrics API. This allows users
+    /// A closure to modify the name and labels used in the Swift Metrics API. This allows users
     /// to overwrite the Metric names in third party packages.
-    public var labelAndDimensionSanitizer: @Sendable (_ label: String, _ dimensions: [(String, String)]) -> (String, [(String, String)])
+    public var nameAndLabelSanitizer: @Sendable (_ label: String, _ dimensions: [(String, String)]) -> (String, [(String, String)])
```
Comment (hamzahrmalik) [inline Sources/Prometheus/PrometheusMetricsFactory.swift:48]: the var is called nameAndLabelSanitizer but the parameters are label, dimensions
this is maybe inconsistent

File: Sources/Prometheus/Docs.docc/swift-metrics.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+    dependencies: [
+      .product(name: "Prometheus", package: "swift-prometheus"),
+      .product(name: "Metrics", package: "swift-metrics"),
+    ]
+  ),
+```
+
+### Setting up Prometheus to export to Swift Metrics
+
+At application startup you need to create a ``PrometheusMetricsFactory`` to register it as
```
Comment (hamzahrmalik) [inline Sources/Prometheus/Docs.docc/swift-metrics.md:39]:
```suggestion
At application startup you need to create a ``PrometheusMetricsFactory`` and register it as
```
?

File: Sources/Prometheus/Docs.docc/index.md [hunk 7]
```diff
@@ -5,15 +5,77 @@ A prometheus client library for Swift.
...
+
+let myCounter = registry.makeCounter(name: "my_counter")
+myCounter.increment()
+
+let myGauge = registry.makeGauge(name: "my_gauge")
+myGauge.increment()
+myGauge.decrement()
+```
+
+Lastly, you can use your ``PrometheusCollectorRegistry`` to generate a Prometheus export as in the
```
Comment (hamzahrmalik) [inline Sources/Prometheus/Docs.docc/index.md:55]: not sure what 'as in' means here

File: Sources/Prometheus/Docs.docc/index.md [hunk 8]
```diff
@@ -5,15 +5,77 @@ A prometheus client library for Swift.
...
+let myCounter = registry.makeCounter(name: "my_counter")
+myCounter.increment()
+
+let myGauge = registry.makeGauge(name: "my_gauge")
+myGauge.increment()
+myGauge.decrement()
+```
+
+Lastly, you can use your ``PrometheusCollectorRegistry`` to generate a Prometheus export as in the 
+text representation:
```
Comment (fabianfett) [inline Sources/Prometheus/Docs.docc/index.md:56]:
```suggestion
Lastly, you can use your ``PrometheusCollectorRegistry`` to generate a Prometheus export in the 
text representation:
```

File: Sources/Prometheus/PrometheusMetricsFactory.swift (hunk 5 above)
Comment (fabianfett) [reply Sources/Prometheus/PrometheusMetricsFactory.swift:48]: this is a bit confusing re Prometheus and Swift Metrics naming.

```
Swift Metrics -> Prometheus
label -> name
dimensions -> labels
```

Comment (fabianfett) [inline Sources/Prometheus/PrometheusMetricsFactory.swift:48]:
```suggestion
    public var nameAndLabelSanitizer: @Sendable (_ name: String, _ labels: [(String, String)]) -> (String, [(String, String)])
```

**Review by hamzahrmalik — APPROVED** [review]
lgtm

**Pull request merged**